src/Json.php

```diff
@@ -103,6 +103,8 @@ public static function encode($valueToEncode, $cycleCheck = false, $options = []
                 return $valueToEncode->toJson();
             } elseif (method_exists($valueToEncode, 'toArray')) {
                 return static::encode($valueToEncode->toArray(), $cycleCheck, $options);
+            } elseif ($valueToEncode instanceof ArrayIterator) {
```

Comment:
@Ocramius I would think Traversable would be too generic as this explicitly targets the "getArrayCopy" method. 
@Ocramius I don't think elseif could be removed since encode would need to process all other objects as it always has. This is just a check to support the ArrayIterator PHP object.